server/src-lib/Hasura/GraphQL/Resolve/Select.hs

```diff
@@ -80,8 +81,16 @@ resolveComputedField computedField fld = fieldAsPath fld $ do
       in case tableArg of
         FTAFirstArgument -> RS.FunctionArgsExp
                             (RS.AETableRow:positional) named
-        FTAName argName  -> RS.FunctionArgsExp positional $
-                            Map.insert (getFuncArgNameTxt argName) RS.AETableRow named
+        FTAName argName index ->
+          let insertAt i a = toList . Seq.insertAt i a . Seq.fromList
+          -- If table argument positional index is less than or equal to
+          -- length of 'positional' arguments then insert the table row
+          -- value in 'positional' arguments else insert the table row
+          -- value with argument name in 'named' arguments
```

Comment:
This seems like it could be a useful function on `FunctionArgsExp` values generally, and it would make this code a little easier to read if it didn’t have to worry about these details. Maybe pull this out into a separate function `insertFunctionArg :: FunctionArgName -> Int -> a -> FunctionArgsExpG a -> FunctionArgsExpG a`, which can go in the same module that defines `FunctionArgsExpG`? The comment you have here can become a Haddock documentation comment on `insertFunctionArg` (and it would be even better if you gave a couple examples of the corresponding SQL in each case!).